ipn/ipnlocal/local_test.go

```diff
@@ -455,6 +455,43 @@ func TestLazyMachineKeyGeneration(t *testing.T) {
 	time.Sleep(500 * time.Millisecond)
 }
 
+func TestZeroExitNodeViaLocalAPI(t *testing.T) {
+	lb := newTestLocalBackend(t)
+
+	// Give it an initial exit node in use.
+	if _, err := lb.EditPrefs(&ipn.MaskedPrefs{
+		ExitNodeIDSet: true,
+		Prefs: ipn.Prefs{
+			ExitNodeID: "foo",
+		},
+	}); err != nil {
+		t.Fatalf("enabling first exit node: %v", err)
+	}
+
+	if _, err := lb.SetUseExitNodeEnabled(true); err != nil {
+		t.Fatal("expected failure")
+	}
+
+	// Zero the exit node
+	pv, err := lb.EditPrefs(&ipn.MaskedPrefs{
```

Comment:
This test passes even if you don't do this. Specifically, if I don't hang on to `pv` and instead use the below on line 487, the test doesn't care about this call to EditPrefs.

If however you change the `true` on line 471 to `false`, then the test becomes meaningful.

```
if got, want := lb.Prefs().InternalExitNodePrior(), tailcfg.StableNodeID(""); got != want {
```